File: 1256-RankTransformOfAnArray/1256-RankTransformOfAnArray.java
// Last updated: 3/26/2026, 1:50:16 AM
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] temp = new int[arr.length];
        for(int i=0;i<arr.length;i++){
            temp[i] = arr[i];
        }
        Arrays.sort(temp);
        int currentRank = 1;
        HashMap<Integer,Integer> rankMap = new HashMap<Integer,Integer>();
        
        for(int i=0;i<arr.length;i++){
            if(!rankMap.containsKey(temp[i])){
                rankMap.put(temp[i],currentRank);
                currentRank++;
            }
            
        }
      
        for(int i=0;i<arr.length;i++){
            arr[i] = rankMap.get(arr[i]);
        }
        return arr;
    }
}

File: 1256-RankTransformOfAnArray/1256-RankTransformOfAnArray.cs
// Last updated: 3/26/2026, 1:50:15 AM
public class Solution {
    public int[] ArrayRankTransform(int[] arr) {
        int[] temp = new int[arr.Length];
        Array.Copy(arr,temp,arr.Length);
        Array.Sort(temp);
        
        Dictionary<int,int> dict = new Dictionary<int,int>();
        int j=1;
        for(int i=0;i<temp.Length;i++)
        {
            if(!dict.ContainsKey(temp[i]))
            {
                dict.Add(temp[i],j);
                j++;
            }
        }
        
        int[] result = new int[temp.Length];
        
        for(int i=0;i<arr.Length;i++)
        {
            result[i] = dict[arr[i]];
        }
        
        return result;
    }
}